### PR TITLE
Jest

### PR DESCRIPTION
**Pull request by Dilden**
This PR aims to incorporate Jest into the workflow.

### CODE REVIEW

File: src/modules/db/mongodb-init.ts
Comment (Dilden) [inline src/modules/db/mongodb-init.ts:9]: I wanted to test this function but had to export it to make it working. Was that proper procedure or would there be a better way to do it? @jimmyhogoboom

Comment (jimmyhogoboom) [reply src/modules/db/mongodb-init.ts:9]: Afaik, that's the only way. It does bring up the argument about whether that function should be tested if it isn't part of the "public" interface of the module. I don't really see the harm in it though.

File: src/modules/db/mongodb-init.ts
Comment (jimmyhogoboom) [inline src/modules/db/mongodb-init.ts:15]: Do we need an explicit type here?

Comment (Dilden) [reply src/modules/db/mongodb-init.ts:15]: Probably not. I was playing around with some things and that change must've snuck in there. Is there any harm in it?

Comment (jimmyhogoboom) [reply src/modules/db/mongodb-init.ts:15]: Shouldn't hurt I don't think, just extraneous. We should probably have a style guide for this project to tell us what to do in these situations.

File: README.md
Comment (jimmyhogoboom) [inline README.md:71]: Instead of asking them to run this command, we could add an npm command to run it for them. Like `test:docker` or something?

Comment (Dilden) [reply README.md:71]: "npm run test" will run the tests inside docker but you still need to have npm installed locally to do that.

Comment (jimmyhogoboom) [reply README.md:71]: Oh wait, that command is what `test` runs. Was the `docker-compose exec node...` command supposed to be something else, or was that not supposed to be in `test`?

Comment (jimmyhogoboom) [reply README.md:71]: oh duh

Comment (jimmyhogoboom) [reply README.md:71]: Wait, could we somehow allow them to run `npm test` inside the container with `docker-compose`?

Comment (Dilden) [reply README.md]: I see what you're saying. Sure, I changed the command `test` to just run the tests and now `test:docker` will run them via the docker containers. Also, updated the readme to reflect those changes.

Comment (jimmyhogoboom) [reply README.md]: Nice work man, this works perfectly

**Comment by Dilden** [comment]
ping @jimmyhogoboom

**Review by jimmyhogoboom — APPROVED** [review]

**Comment by jimmyhogoboom** [comment]
Definitely screwed up and meant to ask for changes, not approve. 🤦‍♂️ 

This is looking great! Nice work getting these tests working. I'd like to know what's going on with the "joined group" test; if that doesn't work correctly, might other tests be giving us false positives? I agree that's something we can tackle later, though.

That said, there are the couple things I want to look at before we merge, but then we're good to go!

File: src/modules/db/index.ts
Comment (Dilden) [reply src/modules/db/index.ts:163]: Ya, that's fair. Must've snuck into my commit.

File: src/utils/__mocks__/telegramUtils.ts
Comment (Dilden) [reply src/utils/__mocks__/telegramUtils.ts:8]: It's been a while since I've looked at it, but remember it was being mocked because it was requiring parts of the framework to work correctly. I didn't want to spend a lot of time mocking all of that as well so created a quick mock that did all I needed.

File: test/integration/handlers.test.js
Comment (Dilden) [reply test/integration/handlers.test.js:86]: Separate tests seem best. But if this test is only testing the group start. There are already other tests to check for a private message from a group admin first.

File: src/modules/db/index.ts
Comment (jimmyhogoboom) [reply src/modules/db/index.ts:163]: @Dilden do we still need to remove this hunk?

Comment (Dilden) [reply src/modules/db/index.ts:161]: Removed in latest commit.

**Comment by Dilden** [comment]
If we're leaving changes in src/modules/db/index.ts (which I think are necessary for the dependency updates), then are we good to merge this?

**Comment by jimmyhogoboom** [comment]
Do it!

**Comment by Dilden** [comment]
Can't merge until it's approvedimage

**Review by jimmyhogoboom — APPROVED** [review]
What the heck, GitHub mobile wasn't showing the status at all, so I thought it was all clear... anyway, merging time!

**Pull request merged**
